src/lib/storage/index/partial_hash/flat_map_iterator.hpp

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include "flat_map_iterator_impl.hpp"
+#include "types.hpp"
+
+/**
+ * The choice of the concrete iterator implementation was subject to a long discussion (see PR 2448). The
+ * PartialHashIndex uses the pointer to implementation (pImpl) idiom, i.e., the non-templated PartialHashIndex instance
+ * holds an opaque pointer to its templated implementation PartialHashIndexImpl. Since the index access is done through
```

Comment:
Maybe add here already what the template parameter is supposed to be (data type)?